src/renderer/components/recording-page/recording-page.test.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-disable no-console */
+/* eslint-disable jest/prefer-expect-assertions */
+
+import { cleanup, fireEvent, render } from '@testing-library/react';
+import React from 'react';
+
+import { noOp } from '../../env-and-consts';
+import mediaService from '../../services/media';
+
+import { RecordingPage } from './index';
+
+jest.mock('../../services/media', () => ({
+  switchOnAudioInput: jest.fn(noOp),
+  stopRecording: jest.fn(noOp),
+  startRecording: jest.fn(noOp),
+  audioBlobToWavArrayBuffer: jest.fn(noOp),
+  createNewAudioGraph: jest.fn(noOp),
+  createOscillator: jest.fn(noOp),
+  playAudioNode: jest.fn(noOp),
+}));
+
+const originalError = console.error;
+
+const flushPromises = () => new Promise(setImmediate);
+const beforeTest = () => {
+  // Supresses compatibility issues with ReactDOM 16.8
+  console.error = (...args: string[]) => {
+    if (/Warning.*not wrapped in act/.test(args[0])) {
+      return;
+    }
+    originalError.call(console, ...args);
+  };
+};
+const afterTest = () => {
+  flushPromises();
+  console.error = originalError;
+  cleanup();
+};
+
+describe('recordingPage', () => {
+  it('starts recording when r is pressed', async () => {
+    beforeTest();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (global as any).__static = '';
+    const alertStartRecording = jest.spyOn(mediaService, 'startRecording');
+    const alertAudioBlobToWavArrayBuffer = jest.spyOn(mediaService, 'audioBlobToWavArrayBuffer');
+
+    render(<RecordingPage onBack={noOp} recordingItems={[]} basePath={''} scaleKey={'C'} octave={2} />);
+    //R is 82
+    fireEvent.keyDown(document, { key: 'R', code: '82' });
```

Comment:
Prefer `userEvent` to `fireEvent`

Reply:
I'm not sure is userEvent has the functionality needed to perform this test.  I need to simulate holding a key down for a few seconds and then releasing it - from what I can see, the closest thing userEvent has is allowing for text to be typed into a input or text element (rather than just typed without focus on a specific element).

Reply:
Sounds good.